querybook/webapp/redux/queryEngine/selector.ts

```diff
@@ -42,16 +50,32 @@ export const queryEngineByIdEnvSelector = createSelector(
         })
 );
 
+export const enabledQueryEngineByIdEnvSelector = createSelector(
+    enabledQueryEngineSelector,
+    (queryEngines) =>
+        new Proxy(arrayGroupByField(queryEngines), {
+            get: (engineById, name: string) =>
+                name in engineById ? engineById[name] : unknownQueryEngine,
+        })
+);
+
 export const queryEngineStatusAndEngineIdsSelector = createSelector(
     queryEngineStatusByIdSelector,
     engineIdsInEnvironmentSelector,
-    (queryEngineStatusById, engineIds) =>
-        engineIds.reduce((pairs, id) => {
-            if (id in queryEngineStatusById) {
-                pairs.push([id, queryEngineStatusById[id]]);
-            }
-            return pairs;
-        }, []) as Array<[number, IQueryEngineStatus]>
+    enabledQueryEngineSelector,
+    (queryEngineStatusById, engineIds, enabledQueryEngines) =>
+        engineIds
+            .filter(
```

Comment:
since there is a filtering layer, we could write this as

```
.filter(id => enabledQueryEngines.some((engine) => engine.id === id) && id in queryEngineStatusById)
.map(id => [id, queryEngineStatusById[id]])
```